commands/project/create.ts

```diff
@@ -28,36 +37,92 @@ exports.describe = uiBetaTag(i18n(`${i18nKey}.describe`), false);
 exports.handler = async options => {
   const { derivedAccountId } = options;
 
-  const hasCustomTemplateSource = Boolean(options.templateSource);
+  let latestRepoReleaseTag;
+  let templateSource = options.templateSource;
+  if (!templateSource) {
+    templateSource = HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH;
+    try {
+      const releaseData = await fetchReleaseData(
+        HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH
+      );
+      if (releaseData) {
+        latestRepoReleaseTag = releaseData.tag_name;
+      }
+    } catch (err) {
+      logger.error(i18n(`${i18nKey}.error.failedToFetchProjectList`));
+      process.exit(EXIT_CODES.ERROR);
+    }
+  }
 
-  let githubRef = '';
+  const projectTemplates = await getProjectTemplateListFromRepo(
+    templateSource,
+    latestRepoReleaseTag || DEFAULT_PROJECT_TEMPLATE_BRANCH
+  );
 
-  if (!hasCustomTemplateSource) {
-    const releaseData = await fetchReleaseData(
-      HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH
-    );
-    githubRef = releaseData.tag_name;
+  if (!projectTemplates.length) {
+    logger.error(i18n(`${i18nKey}.error.failedToFetchProjectList`));
+    process.exit(EXIT_CODES.ERROR);
   }
 
   const { name, template, dest } = await createProjectPrompt(
-    githubRef,
-    options
+    options,
+    projectTemplates
   );
+  const projectName = name || options.name;
+  const projectDest = path.resolve(getCwd(), dest || options.dest);
 
   trackCommandUsage(
     'project-create',
     { type: template.name },
     derivedAccountId
   );
 
-  await createProjectConfig(
-    path.resolve(getCwd(), options.dest || dest),
-    options.name || name,
-    template,
-    options.templateSource,
-    githubRef
+  const {
+    projectConfig: existingProjectConfig,
+    projectDir: existingProjectDir,
+  } = await getProjectConfig(projectDest);
+
+  // Exit if the target destination is within an existing project
+  if (existingProjectConfig && projectDest.startsWith(existingProjectDir)) {
+    logger.error(
+      i18n(`${i18nKey}.errors.cannotNestProjects`, {
+        projectDir: existingProjectDir,
+      })
+    );
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  try {
+    await cloneGithubRepo(templateSource, projectDest, {
+      sourceDir: template.path,
+      tag: latestRepoReleaseTag,
+      hideLogs: true,
+    });
+  } catch (err) {
+    debugError(err);
+    logger.error(i18n(`${i18nKey}.errors.failedToDownloadProject`));
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  const projectConfigPath = path.join(projectDest, PROJECT_CONFIG_FILE);
+
+  const parsedConfigFile: ProjectConfig = JSON.parse(
+    fs.readFileSync(projectConfigPath).toString()
   );
 
+  writeProjectConfig(projectConfigPath, {
+    ...parsedConfigFile,
+    name: projectName,
+  });
+
+  // If the template is 'no-template', we need to manually create a src directory
+  if (template.name === 'no-template') {
```

Comment:
Good candidate for a constant

Reply:
Good call, done in [6970a63](https://github.com/HubSpot/hubspot-cli/pull/1385/commits/6970a63c86ee06d3bb34307d3af05eae6d2a26a2)